test/unit/test_expid.py

```diff
@@ -1,441 +1,73 @@
-    assert "" == experiment_id
-
+from mock import MagicMock
 
-@patch('autosubmit.experiment.experiment_common.db_common')
-def test_create_new_evaluation_experiment_with_empty_current(db_common_mock):
-    current_experiment_id = ""
-    _build_db_mock(current_experiment_id, db_common_mock)
-    experiment_id = new_experiment(_DESCRIPTION, _VERSION, False, False, True)
-    assert "" == experiment_id
+from autosubmit.autosubmit import Autosubmit
+from log.log import AutosubmitCritical
 
 
-def _build_db_mock(current_experiment_id, mock_db_common):
-    mock_db_common.last_name_used = Mock(return_value=current_experiment_id)
-    mock_db_common.check_experiment_exists = Mock(return_value=False)
+@contextmanager
+def does_not_raise():
+    yield
 
 
-@patch('autosubmit.autosubmit.read_files')
-def test_autosubmit_generate_config(read_files_mock, autosubmit):
-    expid = 'ff99'
-    original_local_root_dir = BasicConfig.LOCAL_ROOT_DIR
-
-    with tempfile.TemporaryDirectory() as temp_dir:
-        temp_file_path = os.path.join(temp_dir, 'files')
-        os.makedirs(temp_file_path)
-
-        with tempfile.NamedTemporaryFile(dir=temp_file_path, suffix='.yaml', mode='w') as source_yaml:
-            # Our processed and commented YAML output file must be written here
-            Path(temp_dir, expid, 'conf').mkdir(parents=True)
-            BasicConfig.LOCAL_ROOT_DIR = temp_dir
-
-            source_yaml.write(
-                dedent('''
-                    JOB:
-                        JOBNAME: SIM
-                        PLATFORM: local
-                    CONFIG:
-                        TEST: The answer?
-                        ROOT: No'''))
-            source_yaml.flush()
-            read_files_mock.return_value = Path(temp_dir)
-
-            parameters = {
-                'JOB': {
-                    'JOBNAME': 'sim'
-                },
-                'CONFIG': {
-                    'CONFIG.TEST': '42'
-                }
-            }
-            autosubmit.generate_as_config(exp_id=expid, parameters=parameters)
-
-            source_text = Path(source_yaml.name).read_text()
-            source_name = Path(source_yaml.name)
-            output_text = Path(temp_dir, expid, 'conf', f'{source_name.stem}_{expid}.yml').read_text()
-
-            assert source_text != output_text
-            assert '# sim' not in source_text
-            assert '# sim' in output_text
-            assert '# 42' not in source_text
-            print(output_text)
-            assert '# 42' in output_text
-
-    # Reset the local root dir.
-    BasicConfig.LOCAL_ROOT_DIR = original_local_root_dir
-
-
-@patch('autosubmit.autosubmit.YAML.dump')
-@patch('autosubmit.autosubmit.read_files')
-def test_autosubmit_generate_config_resource_listdir_order(
-        read_files_mock,
-        yaml_mock,
-        autosubmit
-) -> None:
+def build_db_mock(current_experiment_id, mock_db_common):
```

Comment:
In this file, it's probably OK to not have the `build_db_mock`. But if you want to keep the function in case we get more tests that will use it later, then pass the mocker to it, or make it a local fixture, etc.